pilot/pkg/networking/core/v1alpha3/gateway.go

```diff
@@ -406,22 +404,35 @@ func buildGatewayListenerTLSContext(server *networking.Server, enableSds bool) *
 		}
 	}
 
-	var trustedCa *core.DataSource
-	if len(server.Tls.CaCertificates) != 0 {
-		trustedCa = &core.DataSource{
-			Specifier: &core.DataSource_Filename{
-				Filename: server.Tls.CaCertificates,
+	// If SDS is enabled at gateway, credential name is specified, and tls mode is MUTUAL, create SDS config for gateway to fetch
+	// certificate validation context at gateway agent. Otherwise, set up validation context in the original way.
+	if enableSds && server.Tls.CredentialName != "" && server.Tls.Mode == networking.Server_TLSOptions_MUTUAL {
```

Comment:
you dont need the check for MUTUAL TLS. Just fetch from SDS if the credential name is specified.